Fix: Register all BFT sub-protocols during IBFT2→QBFT consensus migration

saeeddawod opened the pull request:
## Summary

This PR fixes a critical bug that prevents successful **IBFT2 → QBFT consensus migration** in Besu. During migration, nodes would stall and fail to produce blocks because they could only speak one BFT wire protocol instead of both.

## Problem

When running a consensus migration from IBFT2 to QBFT using a `transitions` configuration in genesis, the network would randomly stall. The migration worked sometimes by luck, but failed other times.

### Root Cause

In `ConsensusScheduleBesuControllerBuilder.createSubProtocolConfiguration()`, the original code was:

```java
return besuControllerBuilderSchedule
    .get(besuControllerBuilderSchedule.keySet().stream().skip(1).findFirst().orElseThrow())
    .createSubProtocolConfiguration(ethProtocolManager, maybeSnapProtocolManager);
```

**Issues with this approach:**

1. **Non-deterministic ordering**: `besuControllerBuilderSchedule` is a `HashMap`, so `keySet().stream().skip(1)` produces unpredictable results depending on hash ordering
2. **Only one protocol registered**: Only the sub-protocol configuration from ONE consensus mechanism was registered
3. **Random failures**: If HashMap ordering picked QBFT protocol while consensus was still IBFT2, nodes couldn't exchange consensus messages → network stalls. Sometimes by luck it would pick the matching protocol and work.

## Solution

The fix registers **all BFT sub-protocols** from all scheduled consensus mechanisms:

- **IBF/1** (IBFT2) - for pre-migration communication  
- **istanbul/100** (QBFT) - for post-migration communication

This allows nodes to communicate using both protocols during the transition period. The fix:

1. Iterates through all consensus builders (sorted by block number for determinism)
2. Merges their sub-protocol configurations
3. Avoids duplicates (e.g., EthProtocol which is common to both)

## Testing

### Migration Test Results

| Phase | Block | Consensus | Status |
|-------|-------|-----------|--------|
| Pre-fork | #0-49 | IBFT2 | ✅ IbftBesuControllerBuilder |
| Fork | #50 | QBFT | ✅ QbftBesuControllerBuilder |
| Post-fork | #50-100+ | QBFT | ✅ Continuous block production |

### Log Evidence

```
# Pre-fork (IBFT2)
IbftBesuControllerBuilder | Produced #49 / 0 tx / ...

# Fork block (QBFT takes over)  
QbftBesuControllerBuilder | Imported empty block #50 / 0 tx / ...

# Post-fork (QBFT continues)
QbftBesuControllerBuilder | Produced empty block #73 / 0 tx / ...
```

## Checklist

- [x] Checked out [contribution guidelines](https://github.com/hyperledger/besu/blob/main/CONTRIBUTING.md)
- [x] Considered documentation - migration docs may need update
- [x] Considered changelog - This is a bug fix for consensus migration
- [x] No database changes

## Local Tests

- [x] Manual migration test (IBFT2 → QBFT) - **PASSED**
- [ ] spotless: `./gradlew spotlessApply`
- [ ] unit tests: `./gradlew build`
- [ ] acceptance tests: `./gradlew acceptanceTest`

macfarla commented:
@saeeddawod spotless check is failing, and I can't push to your branch to fix it - can you run spotlessApply gradle target. Assigning back to you.

jframe commented on app/src/main/java/org/hyperledger/besu/controller/ConsensusScheduleBesuControllerBuilder.java:
```diff
@@ -225,9 +227,37 @@ protected JsonRpcMethods createAdditionalJsonRpcMethodFactory(
+    // - IBF/1 (IBFT2) for pre-migration communication
+    // - istanbul/100 (QBFT) for post-migration communication
+    final SubProtocolConfiguration mergedConfig = new SubProtocolConfiguration();
+    final java.util.Set<String> addedProtocolNames = new java.util.HashSet<>();
```
nit: be nicer add an import for HashSet so we don't need the fully qualified name

saeeddawod commented:
Thanks @jframe  @macfarla  Addressed the review:
  - Fixed imports (HashSet, Set instead of fully qualified names)
  - Added createsSubProtocolConfigurationWithAllSubProtocols() test
  - Ran spotlessApply
